### PR TITLE
feat(rag): list rag database documents

### PR DESCRIPTION
**Pull request by mhamann**
### **PR Type**
Enhancement

___

### **Description**
- Add document listing functionality to RAG database with pagination support

- Implement `list_documents()` method in ChromaStore for chunk retrieval

- Create new Celery task `rag_list_database_documents_task` for aggregating chunks by source

- Add REST API endpoint `/documents` to expose document listing with filtering

- Register new task module and add comprehensive test coverage

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["API Endpoint<br/>/documents"] -->|calls| B["list_rag_documents<br/>rag_client.py"]
  B -->|triggers| C["rag_list_database_documents_task<br/>Celery Task"]
  C -->|queries| D["ChromaStore<br/>list_documents"]
  D -->|returns chunks| C
  C -->|aggregates by source| E["Document List<br/>with metadata"]
  E -->|returns| A
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>celery_app.py</strong><dd><code>Register stats_tasks module with Celery</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/celery_app.py

<ul><li>Register <code>tasks.stats_tasks</code> module in Celery app configuration for both <br>Redis and filesystem brokers<br> <li> Import <code>tasks.stats_tasks</code> module during app initialization to ensure <br>task registration</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-32c5797c3a8328c00d12d1f665e96484e55da79b1bccd8466637390dd51d5e44">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>chroma_store.py</strong><dd><code>Add document listing with pagination support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/components/stores/chroma_store/chroma_store.py

<ul><li>Add <code>list_documents()</code> method supporting pagination with <code>limit</code> and <br><code>offset</code> parameters<br> <li> Support optional content inclusion via <code>include_content</code> flag<br> <li> Parse metadata and extract source information from chunks<br> <li> Return tuple of Document objects and total count</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-043df4653d74298401e3a46718de09855fc1dfd1ec557c38cab8fb2f50228242">+71/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Export new document listing task</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/tasks/__init__.py

<ul><li>Export new <code>rag_list_database_documents_task</code> from stats_tasks module<br> <li> Add task to <code>__all__</code> list for public API</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-1eeee2ab7a27dc40299138fcc796eba2b53ebf83e1501b26de7b6caa83989823">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>stats_tasks.py</strong><dd><code>Implement document listing Celery task</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/tasks/stats_tasks.py

<ul><li>Implement <code>rag_list_database_documents_task</code> Celery task for listing <br>documents with pagination<br> <li> Aggregate chunks by source file to provide document-level information<br> <li> Extract and normalize metadata fields (date_ingested, file_size, <br>parser_used)<br> <li> Add <code>_extract_filename()</code> helper function to parse source paths</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-81e94871ae99b6d5dcc54e695471d5100c2f4f965d02e0ffb3c1a57eb6a30d43">+169/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>router.py</strong><dd><code>Add document listing REST API endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

server/api/routers/rag/router.py

<ul><li>Add <code>RAGDocumentResponse</code> Pydantic model for document API responses<br> <li> Implement <code>GET /documents</code> endpoint with query parameters for database, <br>limit, and offset<br> <li> Add database validation and default database selection logic<br> <li> Call <code>list_rag_documents()</code> client function and convert results to <br>response model</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-74628d29907b8533bda60f3de5d777718db3763cc6bbc3c4f7dc44b030cd0a4f">+109/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>rag_client.py</strong><dd><code>Add Celery client for document listing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

server/core/celery/rag_client.py

<ul><li>Add <code>list_rag_documents()</code> function to call <code>rag.list_database_documents</code> <br>Celery task<br> <li> Support pagination parameters (limit, offset)<br> <li> Use sync task polling with 60-second timeout and 0.5-second poll <br>interval<br> <li> Return documents list with total count and database name</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-ae047300bef7e05c5108b15a00d9cf062a01e2ef429242f2b07fb5e3cbbd622f">+30/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_chroma_store.py</strong><dd><code>Add ChromaStore list_documents tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/tests/components/stores/test_chroma_store.py

<ul><li>Add 8 test cases for <code>list_documents()</code> method covering empty <br>collections, pagination, metadata preservation<br> <li> Test content inclusion/exclusion and source field preservation<br> <li> Verify Document object structure and total count accuracy</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-4ad410654b047a6aa46aab7699ac83888a1cd222570625a6ac5cc2da0e17a293">+74/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_stats_tasks.py</strong><dd><code>Add tests for document listing task</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rag/tests/test_stats_tasks.py

<ul><li>Add <code>TestExtractFilename</code> class with 6 tests for filename extraction <br>from various path formats<br> <li> Add <code>TestListDocumentsTaskImport</code> class verifying task registration and <br>importability<br> <li> Add <code>TestListDocumentsTaskAggregation</code> class testing chunk aggregation <br>logic<br> <li> Update <code>TestStatsTaskTaskImport</code> to include new task and helper function <br>imports</ul>

</details>

  </td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/547/files#diff-a092d7a660d8efd9082caf848a4d10eed48d0995eb570bf80e782e2c24138a3a">+110/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

> [!NOTE]
> Introduce end-to-end document listing for RAG databases with pagination via ChromaStore, Celery task, FastAPI endpoint, CLI integration, and tests.
> 
> - **RAG Backend**:
>   - **ChromaStore**: Add `list_documents(limit, offset, include_content)` to paginate chunks and return `Document`s with parsed metadata.
>   - **Celery Task**: New `rag.list_database_documents` aggregates chunks by `source` into document-level metadata; includes `_extract_filename` helper.
>   - **Celery App**: Register/import `tasks.stats_tasks` so the new task is loaded.
>   - **RAG Client**: Add `list_rag_documents(project_dir, database, limit, offset)` with polling.
> - **API**:
>   - Add `GET /v1/projects/{namespace}/{project}/rag/databases/{database_name}/documents` returning `RAGDocumentResponse` list with pagination.
> - **CLI**:
>   - `lf rag list`: require `--database`; fetch from `/rag/databases/{db}/documents`; support `--limit` and metadata filters.
> - **Tests**:
>   - ChromaStore: add coverage for `list_documents` (empty, pagination, content on/off, metadata/source preservation).
>   - Tasks: tests for `_extract_filename`, task registration, and aggregation logic; update imports/exports.
> - **Docs/Tooling**:
>   - Update `AGENTS.md` with Nx usage guidelines.
>   - Bump Nx plugin versions in `nx.json` (`@nx/{eslint,vite,jest}` to 22.0.2).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8a8804e25d23fbe5517fab9e8cc68705d6ab3ddb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by mhamann** [comment]
depends on #543 to be merged first

File: server/api/routers/rag/router.py [hunk 1]
```diff
@@ -130,6 +131,114 @@ async def get_rag_stats(
...
+    size_bytes: int = Field(default=0, description="Document size in bytes")
+    parser_used: str = Field(default="unknown", description="Parser used to process")
+    date_ingested: str = Field(default="", description="Date document was ingested")
+    metadata: dict[str, Any] | None = Field(
+        default=None, description="Additional metadata"
+    )
+
+
+@router.get(
+    "/documents",
```
Comment (BobbyRadford) [inline server/api/routers/rag/router.py]: Thoughts on using a /databases/{database_name} path (like we do for other routes in this file) over a query param?

File: server/api/routers/rag/router.py [hunk 2]
```diff
@@ -130,6 +131,114 @@ async def get_rag_stats(
...
+
+    # Validate database exists
+    database_exists = False
+    for db in project_obj.config.rag.databases or []:
+        if db.name == database_name:
+            database_exists = True
+            break
+
+    if not database_exists:
+        raise HTTPException(
```
Comment (BobbyRadford) [inline server/api/routers/rag/router.py]: We have a typed DatabaseNotFoundException you could use

File: server/api/routers/rag/router.py (hunk 1 above)
Comment (mhamann) [reply server/api/routers/rag/router.py]: Yes, I think we should use that here. I opted not to for the `/stats` API, since you could pull stats for all DBs at once. But I don't think we should let you pull docs for all DBs at once.
